DAQ_FW/lib/IADCSensor/IADCSensor.h

```diff
@@ -1,32 +1,36 @@
 #ifndef IADCSENSOR_LIB
 #define IADCSENSOR_LIB
 
+#include <Wire.h>
+#include <Adafruit_ADS1X15.h>
 #include "Logger.h"
 #include "system_config.h"
 
 class IADCSensor
 {
 public:
-    IADCSensor(const char *_SensorName, uint16_t _SensorID, uint16_t _ADC_ID, uint16_t _ADC_GAIN_MODE);
+    IADCSensor(const char *_SensorName, uint16_t _SensorID, uint16_t _ADC_ID);
+
+    // call in setup(), initializes the adc
+    // TODO: may need to be changed as we will have multiple sensors on single adc
+    // needs to be hardware tested
+    void Initialize();
 
     // Read function - to be overwritten by the Driver class
     // When child class is ready, make abstract
-    virtual uint16_t Read() const;
+    virtual int16_t Read();
 
-    float Process() const;
+    float Process();
 
 private:
-    // bit to voltage conversion factor
-    float mVOLTAGE_PER_BIT;
+    Adafruit_ADS1115 mADS;
 
     // SENSOR METADATA:
     // Currently Generic Parameters, to be expanded as needed
     // TODO: add more specific data, depends on the sensors
     const char *mSensorName;
     uint32_t mSensorID;
```

Comment:
To deal with having to remove const for our methods, we should just make this variable const as well. It's only assigned at initialization and is constant afterwards so we can set it in the constructor initialization list

Reply:
Are you talking about the metadata variables here or the mADS object?

Reply:
the metadata variables. from my understanding mADS has to be non const because we call non const methods with it